### PR TITLE
[TASK-7191] [TASK-7155] [TASK-7093] Receive to peanut wallet

### PR DESCRIPTION
**Pull request by jjramirezn**
This PR does the following

- Enables peanut wallet functionality per user flag
- Opens the /request/pay and /claim pages to non users
- Allows a user with a Peanut wallet to generate a request to the peanut wallet
- Shows the correct balance on the user home

Still missing:
- Send money
- Wallet page
- Automatically choose token/chain for peanut wallet users

### CODE REVIEW

File: src/app/(mobile-ui)/home/page.tsx [hunk 1]
```diff
@@ -64,19 +61,24 @@ const Home = () => {
                             marginLeft: -cardMargin,
                         }}
                     >
-                        {wallets.length ? (
+                        {hasWallets ? (
```
Comment (Hugo0) [inline src/app/(mobile-ui)/home/page.tsx:64]: nicer

File: src/app/(mobile-ui)/home/page.tsx [hunk 2]
```diff
@@ -88,99 +90,26 @@ const Home = () => {
...
+                                    <WalletCard
+                                        key={wallet.address}
+                                        type="wallet"
+                                        wallet={wallet}
+                                        username={username}
+                                        selected={selectedWalletIndex === index}
+                                        onClick={() => handleCardClick(index)}
+                                    />
+                                ))}
+
```
Comment (Hugo0) [inline src/app/(mobile-ui)/home/page.tsx]: WalletCard, nice

File: src/components/Home/WalletCard.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import classNames from 'classnames'
+import Link from 'next/link'
+import { formatUnits } from 'viem'
+import PeanutWalletIcon from '@/assets/icons/peanut-wallet.png'
+import Icon from '@/components/Global/Icon'
+import { shortenAddressLong } from '@/utils'
+import { IWallet } from '@/interfaces'
+
+interface WalletCardProps {
+    type: 'wallet' | 'add'
```
Comment (Hugo0) [inline src/components/Home/WalletCard.tsx]: Thought:

Shouldn't we have 3 types?

- peanut smart wallet
- externally connected wallet
- add

And potentially in the future:
- watch wallets and non evm wallets

File: src/components/Home/WalletCard.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        <p className="text-md">
+                            peanut.me/<span className="font-bold">{username}</span>
+                        </p>
+                    </div>
+                    <p className="text-4xl font-black sm:text-5xl">
+                        $ {Number(formatUnits(wallet.balance, 6)).toFixed(2)}
+                    </p>
+                    <div>
+                        <div className="flex flex-col">
+                            <p className="text-xl font-black sm:text-2xl">{shortenAddressLong(wallet.address)}</p>
```
Comment (Hugo0) [inline src/components/Home/WalletCard.tsx:85]: should this be shortened, or AddressLink?

File: src/components/Home/WalletCard.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                                <span className="text-base">Add BYOW wallet</span>
+                            </div>
+                        </Card.Content>
+                    </Link>
+                </Card>
+            </motion.div>
+        )
+    }
+
+    if (!wallet) return null
```
Comment (Hugo0) [inline src/components/Home/WalletCard.tsx]: Q: whats the point of this? why does wallet have to be optional? Wouldnt it be better to force it?

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx [hunk 6]
```diff
@@ -101,7 +101,6 @@ export const AlreadyPaidLinkView = ({ requestLinkData }: { requestLinkData: _con
                         Discord!
                     </a>
                 </label>
-                <PaymentsFooter />
```
Comment (Hugo0) [inline src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx:104]: why no paymentsFooter!

File: src/context/walletContext/walletContext.tsx [hunk 7]
```diff
@@ -138,11 +154,11 @@ export const WalletProvider = ({ children }: { children: ReactNode }) => {
                 addAccount({
                     accountIdentifier: wagmiAddress,
                     accountType: interfaces.WalletProviderType.BYOW,
-                    userId: user?.user.userId as string, // will always be defined, since user logged in
+                    userId: user.user.userId as string, // will always be defined, since user logged in
                 })
             }
         }
-    }, [wagmiAddress])
+    }, [wagmiAddress, wallets, user])
```
Comment (Hugo0) [inline src/context/walletContext/walletContext.tsx:161]: good on adding these

File: src/interfaces/interfaces.ts [hunk 8]
```diff
@@ -352,6 +352,7 @@ interface User {
     bridge_customer_id: string | null
     full_name: string
     telegram: string | null
+    hasPwAccess: boolean
```
Comment (Hugo0) [inline src/interfaces/interfaces.ts:355]: :+1:

File: src/utils/general.utils.ts [hunk 9]
```diff
@@ -1010,7 +1010,3 @@ export async function fetchTokenSymbol(tokenAddress: string, chainId: string): P
     }
     return tokenSymbol
 }
-
-export function isPwa(): boolean {
-    return window.matchMedia('(display-mode: standalone)').matches || (window.navigator as any).standalone === true
```
Comment (Hugo0) [inline src/utils/general.utils.ts:1015]: niceeee

good consolidation

**Review by Hugo0 — APPROVED** [review]

File: src/components/Home/WalletCard.tsx (hunk 3 above)
Comment (jjramirezn) [reply src/components/Home/WalletCard.tsx]: Absolutely correct, right now we don't have much done in terms of external wallet, when I extracted this component we only have placeholders

File: src/components/Home/WalletCard.tsx (hunk 5 above)
Comment (jjramirezn) [reply src/components/Home/WalletCard.tsx]: I agree, modifying

File: src/components/Home/WalletCard.tsx (hunk 4 above)
Comment (jjramirezn) [reply src/components/Home/WalletCard.tsx:85]: Hmmm AddressLink doesn't make much sense here, but maybe show full address or have a way of copying the address would be good.

File: src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx (hunk 6 above)
Comment (jjramirezn) [reply src/components/Request/Pay/Views/GeneralViews/AlreadyPaid.view.tsx:104]: This is wrong, fixing

**Pull request merged**